src/design-system/HelpButtonWithTooltip.tsx

```diff
@@ -19,11 +19,27 @@ const HelpButton = styled.span`
 
 interface Props {
   children: React.ReactNode;
+  placement:
+    | "auto"
+    | "auto-start"
+    | "auto-end"
+    | "top"
+    | "top-start"
+    | "top-end"
+    | "bottom"
+    | "bottom-start"
+    | "bottom-end"
+    | "right"
+    | "right-start"
+    | "right-end"
+    | "left"
+    | "left-start"
+    | "left-end";
```

Comment:
(Occurrence 1 of 3 of this)

---

Suggestion: You can directly reference the allowed types instead:

```suggestion
  placement: PopperProps["placement"]
```

This way, if a future version of Popper adds more placements in the future, we won't have an inconsistent set of allow props. I think it's also ok not reference that type and have our own list of placements, but in that case we should reduce the list to just the placements that we'll need.

---

Note: The way I found this type is by inspecting the definition of "placement" here! (Cmd-click or alt-click depending on your VSCode setting, right-click » Go to Definition)

![](https://user-images.githubusercontent.com/1570168/81374378-f7d99900-90b3-11ea-8386-4367ade0a734.png)